internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -179,6 +189,105 @@ func getOwnerReferenceFrom(obj, owner client.Object) *metav1.OwnerReference {
 	return nil
 }
 
+func (r *Reconciler) callAfterHooks(ctx context.Context, s *scope.Scope) error {
+	if err := r.callAfterControlPlaneInitialized(ctx, s); err != nil {
+		return err
+	}
+
+	if err := r.callAfterClusterUpgrade(ctx, s); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *Reconciler) callAfterControlPlaneInitialized(ctx context.Context, s *scope.Scope) error {
+	/*
+		TODO: Working comment - DELETE AFTER:
+		- If the cluster topology is being created then mark the AfterControlPlaneInitialized hook so that we can call it later.
+	*/
+	if s.Current.Cluster.Spec.InfrastructureRef == nil && s.Current.Cluster.Spec.ControlPlaneRef == nil {
+		if err := hooks.Mark(ctx, runtimehooksv1.AfterControlPlaneInitialized, s.Current.Cluster, r.Client); err != nil {
+			return errors.Wrapf(err, "failed to mark %s hook", runtimecatalog.HookName(runtimehooksv1.AfterControlPlaneInitialized))
+		}
+	}
+
+	if hooks.IsMarked(runtimehooksv1.AfterControlPlaneInitialized, s.Current.Cluster) {
+		for _, condition := range s.Current.Cluster.GetConditions() {
+			// TODO: Should we check for the ControlPlaneInitialized condition or the ControlPlaneReadyCondition?
+			// From the description of the hook it looks like it should be the ControlPlaneReadyCondition - but need to double check.
```

Comment:
I think `ControlPlaneInitializedCondition` because it sounds plausible to me + matches the hook name:
>// ControlPlaneInitializedCondition reports if the cluster's control plane has been initialized such that the
>// cluster's apiserver is reachable and at least one control plane Machine has a node reference.

@fabriziopandini to confirm

Reply:
Resolving as it's a duplicate of the one below